hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -142,13 +142,19 @@ public void accept(MessageSendingResult result) {
                 handleFailedSending(message, result);
                 if (!isTtlExceeded(message) && shouldRetrySending(result)) {
                     retrySingleThreadExecutor.schedule(() -> retrySending(result),
-                            subscription.getSubscriptionPolicy().getMessageBackoff(), TimeUnit.MILLISECONDS);
+                            extractRetryDelay(result), TimeUnit.MILLISECONDS);
                 } else {
                     handleMessageDiscarding(message, result);
                 }
             }
         }
 
+        private long extractRetryDelay(MessageSendingResult result) {
+            return result.isRetryLater()
+                    ? result.getRetryAfterMillis()
+                    : subscription.getSubscriptionPolicy().getMessageBackoff();
```

Comment:
I think that if header Retry-After is absent then we should respect message backoff from policy...